src/providers/cryptoauthlib/aead.rs

```diff
@@ -0,0 +1,154 @@
+                let key_attributes = self.key_info_store.get_key_attributes(&key_triple)?;
+                op.validate(key_attributes)?;
+
+                let aead_param = rust_cryptoauthlib::AeadParam {
+                    nonce: op.nonce.to_vec(),
+                    tag_length: Some(tag_length as u8),
+                    additional_data: Some(op.additional_data.to_vec()),
+                    ..Default::default()
+                };
+
+                let mut plaintext = op.plaintext.to_vec();
+
+                match self.device.aead_encrypt(
+                    get_aead_algorithm(aead_param, &op.alg),
+                    key_id,
+                    &mut plaintext,
+                ) {
+                    Ok(tag) => {
+                        plaintext.extend(tag);
+
+                        Ok(psa_aead_encrypt::Result {
+                            ciphertext: plaintext.into(),
+                        })
+                    }
+                    Err(error) => {
+                        error!("aead_encrypt failed CAL error {}.", error);
+                        Err(ResponseStatus::PsaErrorGenericError)
+                    }
+                }
+            }
+            None => {
+                error!("aead_encrypt failed, algorithm not supported");
+                Err(ResponseStatus::PsaErrorNotSupported)
+            }
+        }
+    }
+
+    pub(super) fn psa_aead_decrypt_internal(
+        &self,
+        app_name: ApplicationName,
+        op: psa_aead_decrypt::Operation,
+    ) -> Result<psa_aead_decrypt::Result> {
+        match get_tag_length(&op.alg) {
+            Some(tag_length) => {
+                let key_triple =
+                    KeyTriple::new(app_name, ProviderId::CryptoAuthLib, op.key_name.clone());
+                let key_id = self.key_info_store.get_key_id::<u8>(&key_triple)?;
+                let key_attributes = self.key_info_store.get_key_attributes(&key_triple)?;
+                op.validate(key_attributes)?;
+
+                let mut ciphertext = op.ciphertext.to_vec();
+                let tag = ciphertext.split_off(ciphertext.len() - tag_length);
+
+                let aead_param = rust_cryptoauthlib::AeadParam {
+                    nonce: op.nonce.to_vec(),
+                    tag: Some(tag),
+                    additional_data: Some(op.additional_data.to_vec()),
+                    ..Default::default()
+                };
+
+                match self.device.aead_decrypt(
+                    get_aead_algorithm(aead_param, &op.alg),
+                    key_id,
+                    &mut ciphertext,
+                ) {
+                    Ok(true) => Ok(psa_aead_decrypt::Result {
+                        plaintext: ciphertext.into(),
+                    }),
+                    Ok(false) => {
+                        error!("aead_decrypt authentication failed");
+                        Err(ResponseStatus::PsaErrorGenericError)
```

Comment:
This should be `PsaErrorInvalidSignature` ([reference](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_aead_decrypt.html#specific-response-status-codes))